### DISCUSSION TITLE
How do I remove these options from the desktop context menu?

### DISCUSSION THREAD

**Discussion by 0-BlackSpectrum-0** [moudey/Shell#173] [Q&A]
![Screenshot 2023-02-27 222854](https://user-images.githubusercontent.com/79910450/221630219-5781ee08-d959-45d3-ae20-1f5c12894709.png)

**Comment by moudey** [comment]
Add this code to `shell.nss` in `static` section
```
item(find='graphics*|nvidia*' vis=0)
```
Save changes, press `CTRL` + `RIGHT-CLICK` or restart Explorer to reload settings.

**Reply by 0-BlackSpectrum-0** [reply]
Thanks

**Reply by moudey** [reply]
Add the `type` property to specify the desktop menu only
```
item(type='desktop' find='graphics*|nvidia*' vis=0)
```

**Reply by Eta-Beta** [reply]
Good morning, then in the desktop context menu and also files I can delete the entries, but how to proceed for folders or other context menus? like when you select a folder ?

**Reply by moudey** [reply]
> Good morning, then in the desktop context menu and also files I can delete the entries, but how to proceed for folders or other context menus? like when you select a folder ?

Handling filter by `type` property

```
// static
item(type="dir|dir.back" find="delete" image=#f00 vis="disabled")

// dynamic
item(type="dir|dir.back" title="test folder" cmd=msg(sel.path))
```

**Reply by Eta-Beta** [reply]
Hello and thanks for the support. As an inexperienced I made many attempts and then I succeeded. Since the items to disable were in all the context menus, I did this:
`item(type='*' find='Aggiungi il/i file selezionato/i per il backup rapido.' vis=0)`  
 I was wrong ?

Now that there are another question, how do you move an item to another submenu? 
![Screenshot 2023-05-13 230209-A](https://github.com/moudey/Shell/assets/104278263/80d443a0-414f-43dc-8579-dc5c4eacbc3b)

GRAZIE INFINITE PER LA TUA GENTILEZZA 🧡

**Reply by moudey** [reply]
It is correct, add it to the static section before any item. 
It is more likely that delete commands will be placed at the top of the static section before the other items.

Move an item to another sub-menu
```
static
{
  item(find='"delete"' menu="more options")
  item(find='"copy"' menu="common")
}

dynamic
{
  menu(title="common")
  {
  }
}
```

**Reply by Eta-Beta** [reply]
I'm sorry but I just can't understand this code. Then the entry to move is DefenderUI add exclusion in "other options" how do i tell the script

**Reply by moudey** [reply]
A menu must be created first in the dynamic section, if it does not exist, to which the items will be moved from the static section.

We created a "common" item in the dynamic section because it does not exist, then we gave a command in the static section to move a specific "copy" item to the "common" menu.

**Reply by Eta-Beta** [reply]
Thanks a lot, now I understand and it will be useful to me, but I wanted to move the items to the "More Options" submenu in the figure that already exists. 
![Screenshot 2023-05-13 230209-B](https://github.com/moudey/Shell/assets/104278263/34b59d06-fdd7-4fde-b3bd-2f7366f59722)

**Reply by moudey** [reply]
```
item(find='DefenderUI add exclusion' menu=title.more_options)
```

**Reply by Eta-Beta** [reply]
DOOOOOO!!!! .. Thanks again for holding my hand up to here. Unfortunately there is the problem that I'm not good with scripts and I don't even know English so I have to use Google translate. I greet you and go to sleep here it is 02:00 AM

**Reply by moudey** [reply]
It seems that you are near me because the time is the same as in Egypt.  Have a good night's sleep.
